### DISCUSSION TITLE
PLACEHOLDER FOR A VALUE THAT WILL DYNAMICALLY FILLED AT RUNTIME

### DISCUSSION THREAD

**Discussion by lifeofsa** [GrapesJS/grapesjs#5459] [Q&A]
Hello thanks for this beautiful library, I just wanted to ask that can I put a place holder inside a grapesjs builder like this {{firstName}} and then filled this with the actual value at runtime? I am also attaching a Screenshot for more clearer view

So assuming that I have an object coming from an API call like this 
obj =  {
"key": "firstName",
"value": "Arthur Melo"
} 

![Q A](https://github.com/GrapesJS/grapesjs/assets/83239811/04ad0ae0-4815-4fef-b6c0-48a56ec592b5)

I want to see the actual value from a placeholder when I click on a preview or deploy my page
like this..!

![Q A 2](https://github.com/GrapesJS/grapesjs/assets/83239811/410b98fd-c278-42c0-b02c-e4b2c64db79e)

**Comment by javadsamiee** [comment]
You can use these commands

```
 editor.on('run:preview', () => {
    editor.setComponents(editor.getHtml().replaceAll('{firstName}', 'John'));
 });

 editor.on('stop:preview', () => {});
```
